nixos/modules/services/web-apps/lemmy.nix

```diff
@@ -224,21 +254,20 @@ in
 
         requires = lib.optionals cfg.database.createLocally [ "postgresql.service" ];
 
-        path = mkIf (cfg.secretFile != null) [ pkgs.jq ];
-
-        # merge the two configs and prevent others from reading the result
+        # substitute secrets and prevent others from reading the result
         # if somehow $CREDENTIALS_DIRECTORY is not set we fail
-        preStart = mkIf (cfg.secretFile != null) ''
+        preStart = mkIf (secrets != {}) ''
           set -u
-          umask 177
-          jq --slurp '.[0] * .[1]' ${lib.escapeShellArg configFile} "$CREDENTIALS_DIRECTORY/secretFile" > ${lib.escapeShellArg mergedConfig}
+          umask u=rw,g=,o=
+          cd "$CREDENTIALS_DIRECTORY"
+          ${utils.genJqSecretsReplacementSnippet cfg.settings substitutedConfig}
         '';
 
         serviceConfig = {
           DynamicUser = true;
           RuntimeDirectory = "lemmy";
           ExecStart = "${cfg.server.package}/bin/lemmy_server";
-          LoadCredential = mkIf (cfg.secretFile != null) "secretFile:${toString cfg.secretFile}";
+          LoadCredential = lib.foldlAttrs (acc: option: data: acc ++ [ "${option}:${toString data.value}" ]) [] secrets;
```

Comment:
@pennae gives some good advice in the matrix room

> the suggestion is good. if someone wants the secret in the store they can do that with a path literal, and if they don't they can use a string literal

> that's the usual approach in modules: make it a path, and it's up to the user to supply a path-like string if they don't want copies into the store

An example of putting secrets into the nix store is VM tests.

Reply:
I think copying secrets in the Nix store is almost pointless since it is world readable and if it is really necessary a store path can be generated manually.
Doing it implicitly could lead to users accidentally copying their secrets to the Nix store.